web-api/src/storage/elastic/filter.rs

```diff
@@ -249,6 +357,110 @@ enum Clause<'a> {
     MustNot(MustNot<'a>),
 }
 
+fn fully_merge_range(mut clause: Vec<Clause<'_>>) -> Vec<Clause<'_>> {
+    fn is_range<'a>(clause: &Clause<'a>) -> Option<&'a DocumentPropertyId> {
+        if let Clause::Range(range) = clause {
+            Some(range.field)
+        } else {
+            None
+        }
+    }
+
+    let mut i = 0;
+    while i < clause.len() {
+        if let Some(field) = is_range(&clause[i]).map(ToOwned::to_owned) {
+            let mut j = i + 1;
+            while j < clause.len() {
+                if is_range(&clause[j]).map_or(false, |f| f == &field) {
+                    let Clause::Range(range_j) = clause.swap_remove(j) else {
+                        unreachable!(/* clause[j] is range */);
+                    };
+                    let Clause::Range(range_i) = &mut clause[i] else {
+                        unreachable!(/* clause[i] is range */);
+                    };
+                    match (&mut range_i.greater, range_j.greater) {
+                        (Some(greater_i), Some(greater_j)) => greater_i.and(greater_j),
+                        (None, greater_j @ Some(_)) => range_i.greater = greater_j,
+                        (Some(_) | None, None) => {}
+                    }
+                    match (&mut range_i.less, range_j.less) {
+                        (Some(less_i), Some(less_j)) => less_i.and(less_j),
+                        (None, less_j @ Some(_)) => range_i.less = less_j,
+                        (Some(_) | None, None) => {}
+                    }
+                } else {
+                    j += 1;
+                }
+            }
+        }
+        i += 1;
+    }
+
+    clause
+}
+
+fn partly_merge_range(mut clause: Vec<Clause<'_>>) -> Vec<Clause<'_>> {
```

Comment:
:wrench: 

see previous comment

potential fix:

```suggestion
fn merge_ranges_in_or_clause(mut clause: Vec<Clause<'_>>) -> Vec<Clause<'_>> {
```